test/Cocoapods/PodfileLockTest.hs

```diff
@@ -68,8 +69,4 @@ spec_analyze = do
   podLockFile <- T.runIO (TIO.readFile "test/Cocoapods/testdata/Podfile.lock")
   T.describe "podfile lock parser" $
     T.it "parses error messages into an empty list" $
-      case runParser findSections "" podLockFile of
-        Left _ -> T.expectationFailure "failed to parse"
-        Right result -> do
-            result `T.shouldContain` [podSection]
-            result `T.shouldContain` [dependencySection]
+      runParser findSections "" podLockFile `T.parseSatisfies` \result -> podSection `elem` result && dependencySection `elem` result
```

Comment:
I'm not sure I like this option better. We lose the failed to parse exception and it feels more difficult to understand what we're asserting on.